src/client/pythonEnvironments/info/interpreter.ts

```diff
@@ -76,5 +78,42 @@ export async function getInterpreterInfo(
 }
 
 export function getInterpreterHash(interpreter: PythonEnvironment | {path: string}){
-    return sha256().update(interpreter.path).digest('hex');
+    const interpreterPath = getNormalizedInterpreterPath(interpreter.path);
+    return sha256().update(interpreterPath).digest('hex');
+}
+
+/**
+ * Sometimes on CI, we have paths such as (this could happen on user machines as well)
+ *  - /opt/hostedtoolcache/Python/3.8.11/x64/python
+ *  - /opt/hostedtoolcache/Python/3.8.11/x64/bin/python
+ *  They are both the same.
+ * This function will take that into account.
+ */
+export function areInterpreterPathsSame(path1: string = '', path2:string = '', ostype = getOSType(), fs?: IFileSystem){
+    const norm1 = getNormalizedInterpreterPath(path1, ostype);
+    const norm2 = getNormalizedInterpreterPath(path2, ostype);
+    return norm1 === norm2 || (fs ? fs.areLocalPathsSame(norm1, norm2) : false);
+}
+/**
+ * Sometimes on CI, we have paths such as (this could happen on user machines as well)
+ *  - /opt/hostedtoolcache/Python/3.8.11/x64/python
```

Comment:
I'm not sure why these are the same? They're the same on your machine, but why everywhere this pattern is found? I would assume on the CI machine these paths are symlinked to each other?

Reply:
> same? They're the same on your machine, but why everywhere this pattern is found? I would assume on the CI machine these paths are symlinked to each other?



Yes, if its `/bin/xyz` and you have `xyz` then they are the same. I.e. python doesn't hvae multiple different interpreter executables in the same directory. They have `python3.5.exe` and python.exe` in the same directory and both are the same.

& `/bin/`  is only used on linux, on windows its always `scripts/....`
Also we're looking at python file names, hence windows will always be excluded as well.

Reply:
> found? I would assume on the CI machine these paths are symlinked to each other?

I think so, i didn't try to figure that out. This is very new, we didn't see this in the past.
Looking back at older runs i couldn't find this. Either ther'es a change in some python package or CI or the python extension is not finding these symlinked items.

Reply:
I guess I'm just nervous this will break something that used to work.

Somebody has say

```
~/mystuff/python
~/mystuff/bin/python
```

And those aren't the same. I think you're saying this isn't normal. You wouldn't have bin/python and python at the same level.

Reply:
> ou wouldn't have bin/python and python at the same level.

Yes. 
Because if `mystuff/python` is the python environment, then you won't have a different version interpreter under `mystuff/bin/python` (thats very very unlikely, from what i know, python doesn't setup interpreters that way).